src/utils/push.js

```diff
@@ -1,128 +1,213 @@
-        notifee.onForegroundEvent(async ({type, detail}) => {
+      } catch (error) {
+        console.error(error);
+      }
+    });
+
+    return () => {
+      notifee.onForegroundEvent(async ({type, detail}) => {
+        try {
           if (type === EventType.PRESS && detail.notification && isSendbirdNotification(detail.notification.data)) {
+            sb.markPushNotificationAsClicked(detail.notification.data);
             navigationRef.current?.navigate('Notifications');
           }
-        }),
-      );
+        } catch (error) {
+          console.error(error);
+        }
+      });
+    };
+  }
+  startOnBackground() {
+    sb.markPushNotificationAsDelivered(data);
+    // Specifies the action to take when the user presses the notification in the background.
+    notifee.onBackgroundEvent(async ({type, detail}) => {
+      try {
+        if (type === EventType.PRESS && detail.notification && isSendbirdNotification(detail.notification.data)) {
+          sb.markPushNotificationAsClicked(detail.notification.data);
+          navigationRef.current?.navigate('Notifications');
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+
+    messaging().setBackgroundMessageHandler(async remoteMessage => {
+      try {
+        if (isSendbirdNotification(remoteMessage.data)) {
+          sb.markPushNotificationAsDelivered(remoteMessage.data);
+          await this.displayNotification(remoteMessage);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  }
+  displayNotification = async message => {
+    try {
+      const sendbird = parseSendbirdNotification(message.data);
+      await notifee.displayNotification({
+        id: String(sendbird.message_id),
+        title: `${sendbird['push_title'] || sendbird.channel.name || 'Message received'}`,
+        body: sendbird.message,
+        data: message.data,
+        android: {
+          channelId: this.ANDROID_NOTIFICATION_CHANNEL_ID,
+          importance: 4,
+          largeIcon: sendbird.sender?.profile_url || sendbird.channel.channel_url,
+          circularLargeIcon: true,
+          pressAction: {id: 'default'},
+          showTimestamp: true,
+          timestamp: sendbird.created_at,
+        },
+      });
+    } catch (error) {
+      console.error(error);
     }
+  };
+}
+class iOSNotificationHandler {
+  startOnAppOpened() {
+    PushNotificationIOS.getInitialNotification().then(async notification => {
+      try {
+        const data = notification?.getData();
+        const isClicked = data?.userInteraction === 1;
+        const localData = JSON.parse(await AsyncStorage.getItem('loginInformation'));
+        if (data) {
+          sb.markPushNotificationAsDelivered(data);
 
-    if (Platform.OS === 'ios') {
-      PushNotificationIOS.addEventListener('notification', async notification => {
+          if (isClicked && isSendbirdNotification(data)) {
+            sb.markPushNotificationAsClicked(data);
```

Comment:
When the `markPushNotificationAsXXXX` method is called, the SDK needs to send the stored push token to the server.

If the SDK is not logged in, the `sdk.initializeCache(userId)` method needs to be called first to properly retrieve the stored push token from the cache. (SDK's login methods (connect, authenticateFeed) include this cache initialization process.)

for example
```ts
if (!sb.currentUser) {
  const userId = data.sendbird?.recipient?.id ?? loginInformation.userId;
  if (userId) await sdk.initializeCache(userId);
}

sb.markPushNotificationAsClicked(data);
```